tests/codegen/reg-struct-return.rs

```diff
@@ -0,0 +1,63 @@
+// Checks how `reg-struct-return` flag works with different calling conventions:
+// Return struct with 8/16/32/64 bit size will be converted into i8/i16/i32/i64
+// (like abi_return_struct_as_int target spec).
+// x86 only.
+
+//@ compile-flags: --target i686-unknown-linux-gnu -Zreg-struct-return -O -C no-prepopulate-passes
+//@ needs-llvm-components: x86
+
+#![crate_type = "lib"]
+#![no_std]
+#![no_core]
+#![feature(no_core, lang_items)]
+
+#[lang = "sized"]
+trait Sized {}
+#[lang = "copy"]
+trait Copy {}
+
+#[repr(C)]
+pub struct Foo {
+    x: u32,
+    y: u32,
+}
```

Comment:
We should test this slightly more exhaustively. I'd like to see
- single-field structs, too
- structs with fields that aren't valid for all bitpatterns, like, say, `{ bool, bool, i16 }`
- oddly-shaped structs that don't exactly match the sizes we're looking for, but are under the 64-bit limit, like e.g. `{ char, bool, u8 }`
- larger-than-64-bits structs

Also, float handling is consistently weird, so I'd like to see:
- floats, so specifically `struct SinglePair { f32, f32 }`, `Double { f64 }`, and `Single { f32 }`
- another test that disables SSE (but leaves x87 on) and enables this feature
- another test that uses only soft floats and enables this feature

feel free to use revisions or additional test files as you prefer.